### ISSUE TITLE
Diffusion Refinement Plot

### ISSUE THREAD

**Issue by jds8** [jds8/SAE4DLM-CE#44]
The previous entropy plots showed no discernible pattern as to which tokens were unmasked first or had low entropy. Let's check if there are any tokens with a delay before unmasking.

First, let's figure out exactly how tokens are unmasked in diffugpt-s. Is it deterministic? Does a masked token unmask as the token with highest probability in the final layer? 

Second, for every text sequence in the histories you generated, run the model on it and extract the final layer, which gives the logits (if the layer doesn't sum to 1 and gives the softmaxed prob. dist. if the layer does sum to 1) over tokens, and see what the resultant text sequences look like. 

Third, count the proportion of correct tokens and plot how that increases over time.

This gives an indication of how much refinement the diffusion process is doing. Is the refinement process steady and linear or does it do more refinement during certain parts of the diffusion process?
